apps/server/src/modules/provisioning/strategy/oidc/oidc.strategy.ts

```diff
@@ -31,13 +31,15 @@ export abstract class OidcProvisioningStrategy extends ProvisioningStrategy {
 			);
 
 			if (data.externalGroups) {
+				let groups: ExternalGroupDto[] = data.externalGroups;
+
+				if (Configuration.get('FEATURE_VIEW_PROVISIONING_OPTIONS_ENABLED')) {
+					groups = await this.oidcProvisioningService.filterExternalGroups(groups, school?.id, data.system.systemId);
+				}
+
 				await Promise.all(
-					data.externalGroups.map((externalGroup) =>
-						this.oidcProvisioningService.provisionExternalGroup(
-							externalGroup,
-							data.externalSchool,
-							data.system.systemId
-						)
+					groups.map((group: ExternalGroupDto) =>
```

Comment:
When the feature flag is turned of we provision every group. This is the request behavior? Maybe we should ask before someone expected some other...